File: Dia03/d3p1/src/main.py
with open('../../input.txt') as f:
	lines =  f.readlines()
	gamma = '0' * len(lines[0].strip())
	epsilon = '1' * len(gamma)
	count = 0 
	for i in range(len(gamma)):
		for line in lines:
			if line[i] == '1':
				count+=1
				# print(count, len(lines))
				if count>=len(lines)/2:
					gamma = gamma[:i] + '1' + (gamma[i+1:] if i+1 < len(gamma) else '')
					epsilon = epsilon[:i] + '0' + (epsilon[i+1:] if i+1 < len(epsilon) else '')
					break
		# print(gamma)
		count = 0
	print("gamma: " + gamma)
	print("epsilon: " + epsilon)
	print("gamma: " + str(int(gamma,2)))
	print("epsilon: " + str(int(epsilon,2)))
	print('>>> ' + str(int(gamma,2) * int(epsilon,2)))

File: Dia03/d3p1/src/main.rs
fn main(){
	let lines: Vec<_> = include_str!("../../input.txt").lines().collect();

	let mut gamma = vec!["0"; lines[0].len()];
	let mut epsilon = vec!["1"; gamma.len()];
	let mut count = 0;


	for i in 0..gamma.len() {
		for line in &lines {
			if line.chars().nth(i).unwrap() == '1' {
				count += 1;
				if count >= lines.len() / 2 {
					gamma[i] = "1";
					epsilon[i] = "0";
					break
				}
			}
		}
		// console.log(gamma)
		count = 0;
	}
	let gamma = usize::from_str_radix(&gamma.join(""), 2).unwrap();
	let epsilon = usize::from_str_radix(&epsilon.join(""), 2).unwrap();
	println!("gamma: {}", gamma);
	println!("epsilon: {}", epsilon);
	println!(">>> {}", gamma * epsilon);
}